File: backend/src/main/java/com/bookbook/domain/report/entity/Report.java
package com.bookbook.domain.report.entity;

import com.bookbook.domain.report.enums.ReportStatus;
import com.bookbook.domain.user.entity.User;
import com.bookbook.global.entity.BaseEntity;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;

import java.time.LocalDateTime;

@Entity // JPA 엔티티
@Getter
@Setter
@NoArgsConstructor
public class Report extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @CreatedDate
    private LocalDateTime createdDate;

    @LastModifiedDate
    private LocalDateTime modifiedDate;

    private LocalDateTime reviewedDate;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ReportStatus status;

    @ManyToOne
    @JoinColumn(name = "reporter_user_id", nullable = false)
    private User reporterUser;

    @ManyToOne
    @JoinColumn(name = "target_user_id", nullable = false)
    private User targetUser;

    @ManyToOne
    @JoinColumn(name = "closer_id")
    private User closer;

    private String reason;

    public Report(User reporterUser, User targetUser, String reason) {
        this.reporterUser = reporterUser;
        this.targetUser = targetUser;
        this.reason = reason;
        this.status = ReportStatus.PENDING;
    }

    public void markAsReviewed() {
        this.status = ReportStatus.REVIEWED;
        this.reviewedDate = LocalDateTime.now();
    }

    public void markAsProcessed(User closer) {
        this.status = ReportStatus.PROCESSED;
        this.closer = closer;
    }
}

File: backend/src/main/java/com/bookbook/domain/report/entity/Report.kt
package com.bookbook.domain.report.entity

import com.bookbook.domain.report.enums.ReportStatus
import com.bookbook.domain.user.entity.User
import com.bookbook.global.jpa.entity.BaseEntity
import jakarta.persistence.*
import lombok.NoArgsConstructor
import java.time.LocalDateTime

@Entity
@NoArgsConstructor
@AttributeOverride(name = "id", column = Column(name = "report_id"))
class Report(
    reporterUser: User,
    targetUser: User,
    reason: String
) : BaseEntity() {

    var reviewedDate: LocalDateTime? = null

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    var status: ReportStatus = ReportStatus.PENDING

    @ManyToOne
    @JoinColumn(name = "reporter_user_id", nullable = false)
    var reporterUser: User = reporterUser

    @ManyToOne
    @JoinColumn(name = "target_user_id", nullable = false)
    var targetUser: User = targetUser

    @ManyToOne
    @JoinColumn(name = "closer_id")
    var closer: User? = null

    var reason: String = reason

    fun markAsReviewed() {
        this.status = ReportStatus.REVIEWED
        this.reviewedDate = LocalDateTime.now()
    }

    fun markAsProcessed(closer: User) {
        this.status = ReportStatus.PROCESSED
        this.closer = closer
    }
}